crypto/hash.go

```diff
@@ -41,8 +41,14 @@ func NewHash() hash.Hash {
 
 // HashAll takes a set of objects as input, encodes them all using the encoding
 // package, and then hashes the result.
-func HashAll(objs ...interface{}) Hash {
-	return HashBytes(encoding.MarshalAll(objs...))
+func HashAll(objs ...interface{}) (hash Hash) {
+	h := NewHash()
+	enc := encoding.NewEncoder(h)
+	for _, obj := range objs {
+		enc.Encode(obj)
+	}
+	h.Sum(hash[:0])
+	return
```

Comment:
huh. This is both faster and bit-for-bit identical? I would probably feel better with a test or two that just compares some fixed inputs to some expected hash outputs. Not too difficult to set that up as a table test I think.